firebase-perf/src/test/java/com/google/firebase/perf/metrics/AppStartTraceTest.java

```diff
@@ -228,14 +229,24 @@ public void testStartFromBackground() {
   }
 
   @Test
-  public void testFirebasePerfProviderOnAttachInfoUpdatesPerfSession() {
+  public void testFirebasePerfProviderOnAttachInfo_initializesGaugeCollection() {
+    com.google.firebase.perf.session.PerfSession originalPerfSession =
```

Comment:
Why do you need to save the originalPerfSession and set it in the end?

Reply:
Otherwise `AppStartTraceTest.testLaunchActivity()` would fail because `trace.onActivityResumed` calls `SessionManager.getInstance().perfSession().build()`, which needs a real perfSession.

Reply:
Unit tests should not depend on each other. Could you create a clearInstance() method so that every test will initialize a new sessionManager
Reference: https://github.com/firebase/firebase-android-sdk/blob/59e027ae2943d51e061f0fd6bf6208fd5e895267/firebase-perf/src/test/java/com/google/firebase/perf/application/AppStateMonitorTest.java#L89